docs/functions.md

```diff
@@ -0,0 +1,209 @@
+    (error) [... 'spam.error: Can not run more then 1 executions in a single script']
+    ```
+
+!!! important "Execution is always asynchronous"
+    Batch functions are **always executed asynchronously** by the RedisGears engine. That means that they are run in a background thread rather than by the main process of the Redis server.
+
+**Python API**
+```python
+class GearsBuilder.run(arg=None, convertToStr=True, collect=True)
+```
+
+_Arguments_
+
+* _arg_: An optional argument that's passed to the reader as its _defaultArg_. It means the following:
+    * A glob-like pattern for the [KeysReader](readers.md#keysreader) and [KeysOnlyReader](readers.md#keysonlyreaders) readers
+    * A key name for the [StreamReader](readers.md#streamreader) reader
+* _convertToStr_: when `True` adds a [map](operations.md#map) operation to the flow's end that stringifies records
+* _collect_: when `True` adds a [collect](operations.md#collect) operation to flow's end
+
+**Examples**
+```python
+# Runs the function
+GB() \
+  .run()
+```
+
+### Register
+The **Register** action registers a function as an event handler. The function is executed each time an event arrives. Each time it is executed, the function operates on the event's data and once done is suspended until its future invocations by new events.
+
+**Python API**
+```python
+class GearsBuilder.register(regex='*', mode='async', batch=1, duration=0,
+  eventTypes=None, keyTypes=None, onRegistered=None, onFailedPolicy="continue",
+  onFailedRetryInterval=1)
+```
+
+_Arguments_
+
+* _regex_: An optional argument that's passed to the reader as its _defaultArg_. It means the following:
+    * A key prefix for the [KeysReader](readers.md#keysreader) and [KeysOnlyReader](readers.md#keysonlyreaders) readers
+    * A key name for the [StreamReader](readers.md#streamreader) reader
+* _mode_: the execution mode of the triggered function. Can be one of:
+    * **'async'**: execution will be asynchronous across the entire cluster
+    * **'async_local'**: execution will be asynchronous and restricted to the handling shard
+    * **'sync'**: execution will be synchronous and local
+* _batch_: the batch size that triggers execution for the [StreamReader](readers.md#streamreader) reader
+* _duration_: the interval between executions for the [StreamReader](readers.md#streamreader) reader (takes precedence over the _batch_ argument)
+* _eventTypes_: A whitelist of event types that trigger for the [KeysReader](readers.md#keysreader) or [KeysOnlyReader](readers.md#keysonlyreaders) readers. The list may contain one or more:
+    * Any Redis or Redis module command?
+    * Events ...?
+* _keyTypes_: A whitelist of key types that trigger execution for the [KeysReader](readers.md#keysreader) or [KeysOnlyReader](readers.md#keysonlyreaders) readers. The list may contain one or more from the following:
+    * Redis core types: 'string', 'hash', 'list', 'set', 'zset' or 'stream'
+    * Any module data type?
```

Comment:
To get module datatype one need to specify 'module' in the list. You can not filter by specific module datatype here

Reply:
Fixed via 079f280